### DISCUSSION TITLE
How to upload a large file

### DISCUSSION THREAD

**Discussion by bkraeling** [remix-run/remix#10253] [Q&A]
We have a Remix app where we have a route that uploads a video file to S3 using the AWS SDK. Generally, the flow we have works. It uses `unstable_parseMultipartFormData` and an `UploadHandler` function. We are seeing some errors related to large files.

We've confirmed that it isn't S3 throwing the error, or Remix itself, since there aren't any limitations on request size or response time. We've been able to upload our large file (10 GB+) on a very fast connection, but when the connection is slower it doesn't finish. The culprit looks to be the browser timeout, which times out at exactly 5 minutes. 

What's the best way to approach this? If we upload it via the server, we get the timeout. Is there a way to initiate the upload via the client and prevent the browser from killing the request after 5 minutes?

Thanks for your help in advance.

**Comment by akomm** [comment]
When you use the form / fetcher API of remix, its waiting for the response to refresh the loaders. If it takes so long or some other submission happens before, the previous is invalidated. Not saying its because of some other submission afterward, but just want to point out that long-living connection via the remix form / fetch API is maybe bad idea. Instead use web fetch *or* if you want to track progress, you might want to use xmlhttprequest.

**Reply by akomm** [reply]
And obviously: net issues can't be solved if they are responsible for the issue. You can only *handle* them by reacting to events and notifying the user.
